pkg/controller/sshd/sshd_controller.go

```diff
@@ -378,6 +452,45 @@ func newSSHDDeployment(cr *cloudingressv1alpha1.SSHD, configMapList *corev1.Conf
 			MountPath: filepath.Join(authorizedKeysMountPath, volumeName),
 		})
 	}
+	// Sort volume slices by name to keep the sequence stable.
+	sort.Slice(volumes, func(i, j int) bool {
+		return volumes[i].Name < volumes[j].Name
+	})
+	sort.Slice(volumeMounts, func(i, j int) bool {
+		return volumeMounts[i].Name < volumeMounts[j].Name
+	})
+
+	volumeName := hostKeysSecret.ObjectMeta.Name
+	volumes = append(volumes, corev1.Volume{
+		Name: volumeName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName:  hostKeysSecret.ObjectMeta.Name,
+				DefaultMode: pointer.Int32Ptr(0600),
```

Comment:
```suggestion
				DefaultMode: pointer.Int32Ptr(0400),
```
I think you want `0400` to prevent writes inside the pod.